src/Tribe/Editor/Objects/Event.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Models an event (the post type) in the context of the Block Editor.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Editory\Objects
+ */
+
+namespace Tribe\Events\Editor\Objects;
+
+use Tribe__Events__Main as TEC;
+use Tribe__Utils__Array as Arr;
+
+/**
+ * Class Event
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Editory\Objects
+ */
+class Event implements Editor_Object_Interface {
+	/**
+	 * The event data in the format required by the block editor.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string,mixed>
+	 */
+	protected $data;
+	
+	/**
+	 * The post object to model the data on.
+	 *
+	 * @since TBD
+	 *
+	 * @var \WP_Post
+	 */
+	protected $post;
+
+	/**
+	 * Event constructor.
+	 *
+	 * @since TBD
+	 *
+	 * @param int|\WP_Post|null $event The event post ID or object, or `null` to use the global `post` object.
+	 */
+	public function __construct( $event = null ) {
+		$event_candidate = null !== $event ? $event : \tribe_get_request_var( 'post', false );
+		$this->post      = $event_candidate ? \get_post( $event_candidate ) : null;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function data( $key = null, $default = null ) {
+		if ( null === $this->data ) {
+			$this->data = [
+				'is_new_post' => true,
+			];
+
+			if ( $this->post instanceof \WP_Post && TEC::POSTTYPE === $this->post->post_type ) {
+				$meta = Arr::flatten( (array) \get_post_meta( $this->post->ID ) );
+				$post_id = $this->post->ID;
```

Comment:
:warning: Equals sign not aligned with surrounding assignments; expected 4 spaces but found 1 space

<sub>hash: 5610b8370f20c69cb6482580c3ef7c64</sub>